Userland/Libraries/LibWeb/Loader/ResourceLoader.cpp

```diff
@@ -250,6 +250,12 @@ void ResourceLoader::load(LoadRequest& request, Function<void(ReadonlyBytes, Has
             return;
         }
         m_active_requests.set(*protocol_request);
+
+        protocol_request->on_progress = [request, &protocol_request, timeout](Optional<u32>, u32) {
```

Comment:
What if there never is progress in the first place though? I feel like this is something that should be handled by a timer instead...

Reply:
Agree. I've added a timer into the mix. Felt a bit hackish the way I went around to make it work, let me know if you think there's some other more proper way I could go about this.